lib/src/macro.dart

```diff
@@ -272,3 +278,16 @@ abstract class FieldDefinitionBuilder implements DefinitionBuilder {
   void withGetterSetterPair(Declaration getter, Declaration setter,
       {List<Code>? supportingDeclarations});
 }
+
+/// Api used by [DefinitionMacros] that want to annotate classes and retrieve
+/// builders for any declaration within the class.
+///
+/// This allows a macro to exist on a class, but manipulate multiple
+/// declarations within the class.
+abstract class ClassDefinitionBuilder implements DefinitionBuilder {
+  ConstructorDefinitionBuilder constructorBuilder(String name);
+
+  FieldDefinitionBuilder fieldBuilder(String name);
+
+  FunctionDefinitionBuilder methodBuilder(String name);
```

Comment:
These look great. I would probably name them more like verb-like `addConstructor()`, `addField()`, `addMethod()`. Or maybe `define` instead of `add`?

Reply:
Well, these methods don't actually add the field or method - it has to be pre-existing. They just give you the builder object which you can then use to implement it.